datafusion/physical-plan/src/joins/nested_loop_join.rs

```diff
@@ -238,6 +238,19 @@ impl NestedLoopJoinExec {
 
         PlanProperties::new(eq_properties, output_partitioning, mode)
     }
+
+    fn maintains_input_order(join_type: JoinType) -> Vec<bool> {
```

Comment:
Is there a benefit having two `maintains_input_order()` in both `ExecutionPlan` implementation and struct implementation, wdyt? If not, we could move this logic directly into `ExecutionPlan` implementation.

Reply:
`maintains_input_order` is also called in `try_new` before the `NestedLoopJoinExec` is constructed. The `ExecutionPlan` implementation takes `self` as a parameter, and thus cannot be called in `try_new`